### ISSUE TITLE
Issue 14 - Build expression classifier (4-class: engaged, confused, bored, neutral)

### ISSUE THREAD

**Issue by bipulshahi** [newton-school-ai/engageiq-ai_2#14]
## Why

Facial expressions provide direct evidence of cognitive state. A confused expression means the student does not understand the current material - information that no other signal (gaze, pose, drowsiness) can provide. A bored expression combined with correct gaze (looking at screen) reveals a student who is physically present but mentally checked out.

The 4-class simplification (engaged, confused, bored, neutral) from FER2013's standard 7 classes (anger, disgust, fear, happy, sad, surprise, neutral) is deliberate. Anger and disgust are irrelevant in a classroom context. Fear and surprise are too rare to be useful. Happy maps to engaged. Sad maps to bored. This reduces the classification problem while keeping the educationally relevant classes.

Transfer learning from a pre-trained FER model is the practical approach. Training from scratch would require a labeled classroom expression dataset that does not exist. Fine-tuning an existing FER model on the 4 target classes gives reasonable accuracy (70%+) with minimal training data.

## What needs to be built

A facial expression classifier that categorizes face crops into 4 engagement-relevant classes using transfer learning.

## Files to create or update

- `src/detection/expression.py` - Expression enum and `ExpressionClassifier` class

## How this affects overall development

Expression contributes 25% weight to the engagement score (#16). The CONFUSED state in the state machine (#17) is primarily triggered by the expression classifier. The intervention agent (#29) uses expression data to identify confusing content segments. If the classifier has low accuracy, confused students are not identified and the system misses its most unique signal.

## How to test locally

```bash
# Demo with webcam
python -m src.detection.expression --demo
# Should show expression label updating in real-time
# Try: smile (engaged), frown (confused), blank stare (bored), resting face (neutral)

python -c "
from src.detection.expression import ExpressionClassifier, Expression
import cv2

classifier = ExpressionClassifier()
face_crop = cv2.imread('tests/fixtures/sample_face.jpg')
result = classifier.classify(face_crop)
print(f'Expression: {result.expression}, Confidence: {result.confidence:.2f}')
# Example: Expression.NEUTRAL, 0.78
"

pytest tests/test_expression.py -v
```

## Acceptance Criteria

- 4 expression classes: ENGAGED, CONFUSED, BORED, NEUTRAL
- Uses pre-trained FER model with transfer learning for 4-class output
- Accuracy > 70% on FER2013 test set (mapped to 4 classes)
- Inference time < 50ms per face crop on CPU
- Returns expression enum + confidence score (0-1)
- Handles poor lighting gracefully (returns NEUTRAL with low confidence, not crash)
- At least 4 tests: one per expression class with sample images

## Branch

`feature/issue-14-expression`

## Depends on

Closes #8
